shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -277,7 +289,11 @@ private void setSystemChromeSystemUIOverlayStyle(
   }
 
   private void popSystemNavigator() {
-    activity.finish();
+    if (fragmentActivity != null) {
+      fragmentActivity.getOnBackPressedDispatcher().onBackPressed();
```

Comment:
This will create a problem if a FlutterFragment subclass uses the OnBackPressedDispatcher to receive onBackPressed callbacks, which is a logical alternative to explicitly calling through from the activity (my app has already done this). I think that this change needs to account for that, and the best way is to have FlutterFragment's onBackPressed method explicitly triggered by an OnBackPressedCallback so that the Flutter framework can disable the callback in this specific case.

Reply:
ya, if I understood your comment (I think you're just saying we should do https://github.com/flutter/flutter/issues/67011), I agree these are related (https://github.com/flutter/engine/pull/22495#issuecomment-726936914). 

Though these things also don't 100% overlap. Whether FlutterFragment implements OnBackPressedCallback or not, you can still imperatively call SystemNavigator.pop() from Dart code with or without a back button press from the Android side. So we should still handle this somehow.

Reply:
Yes, that's true and I agree. This change is absolute a necessary component to fixing flutter/flutter#67011

Reply:
Agree that this doesn't substitute or alleviate the need to fix flutter/flutter#67011.

Reply:
If we did flutter/flutter#67011 i think it will be breaking change and we must update the [doc ](https://flutter.dev/docs/development/add-to-app/android/add-flutter-fragment?tab=forward-activity-calls-java-tab) @xster wdyt ?

[Adding a Flutter Fragment to an Android app Doc ](https://flutter.dev/docs/development/add-to-app/android/add-flutter-fragment?tab=forward-activity-calls-java-tab)

the onBackPressed Override will not be invoked
```
@Override
 public void onBackPressed() {
   flutterFragment.onBackPressed();
 }
```

Reply:
You could probably make it an opt-in argument to `NewEngineFragmentBuilder` to avoid a breaking change, but I don't know if you'd want to.

Reply:
**Suggestion**

_Usecase_ : If a `FlutterFragment` subclass uses the `OnBackPressedDispatcher`

```
public void popSystemNavigator() {

..........................................
..........................................

  OnBackPressedDispatcher onBackPressedDispatcher = ((ComponentActivity) activity).getOnBackPressedDispatcher();

  if (!onBackPressedDispatcher.hasEnabledCallbacks()) {
    onBackPressedDispatcher.onBackPressed();
    return;
  }

  FragmentManager fragmentManager = ((ComponentActivity) activity).getSupportFragmentManager();

  if (fragmentManager.getBackStackEntryCount() > 0) {
    getActivity().getSupportFragmentManager().popBackStack();
  } else {
    getActivity().finish();
  }
}
```

Reply:
That doesn't quite work either because there may be other enabled callbacks besides the FlutterFragment's.

This issue still hasn't been addressed. As written, this PR will break my app.

Reply:
Can you please provide a test case that has to pass ?

Reply:
Here's an example App implementation: https://github.com/math1man/FlutterFragment/tree/master/android/app/src/main/java/com/example/fragmentbug
It's worth noting that due to the async nature of Dart-Native communication, the app won't actually stack-overflow but instead just loop forever between Dart and Native.

I'll see if I can provide a sample test case too.

Reply:
This PR will not be merged until you provide the test case . If you think you can create new PR for your case you are welcome.

Reply:
For some reason I can't run the tests locally, but https://github.com/flutter/engine/pull/22552 includes a test case that I believe should cover the use case I'm describing.